force-app/main/default/classes/CheckPayment.cls

```diff
@@ -1,74 +1,28 @@
 /* ------------ Used in OOP Basics Assignment ------------ */
 
-public class CheckPayment {
+public class CheckPayment extends Payment {
 
-    //Static Variables to hold the various messages we use
-    public static final String SUCCESS_MESSAGE = 'Success!';
-    public static final String INVALID_AMOUNT = 'Amount should be greater than 0';
-    public static final String INVALID_ACCOUNT_NUMBER = 'Account number is invalid';
-    public static final String MISSING_PAYMENT_INFO = 'Payment details are missing';
     public static final String MASKED_ACCOUNT_ERROR = 'You cannot retrieve the masked checking account number until you set a value';
-
-    //Use Static initilization to set our Valid Account Number Length
-    private final Integer validAccountNumberLength;
     {
-        validAccountNumberLength = 10;
-    }
-
-    //Check for a Valid Account Number Length when the value gets set
-    private String accountNumber {
-        get;
-        set {
-            accountNumber = value;
-            if (accountNumber.length() != validAccountNumberLength) {
-                throw new CheckPaymentException(INVALID_ACCOUNT_NUMBER);
-            }            
-        }
-    }
-
-    //Check for a Valid Amount when the value gets set
-    private Decimal amount {
-        get;
-        set {
-            amount = value;
-            if (amount < 0) {
-                throw new CheckPaymentException(INVALID_AMOUNT);
-            }
-        }        
-    }
-
-    //Constructs a message with the payment information
-    private String transactionMessage {
-        set; 
-        get {
-            return SUCCESS_MESSAGE + ' A payment of ' + this.amount + ' has been applied from account ' + getMaskedAccountNumber();
-        }
+        this.validAccountNumberLength=10;
     }
 
     //Sets our class variables based on local variables
-    public void setPaymentDetails(String accountNumber, Decimal amount) {
+    override public void setPaymentDetails(String accountNumber, Decimal amount) {
```

Comment:
should be public override , not override public

Reply:
done